motor/discovery/aws/ecr.go

```diff
@@ -77,61 +79,47 @@ func (a *EcrImages) getRepositories() []*jobpool.Job {
 			clonedConfig := a.config.Copy()
 			clonedConfig.Region = region
 			svc := ecr.NewFromConfig(clonedConfig)
-			imgs := []*asset.Asset{}
+			assets := []*asset.Asset{}
 
 			repoResp, err := svc.DescribeRepositories(ctx, &ecr.DescribeRepositoriesInput{})
 			if err != nil {
-				return imgs, nil
+				return assets, nil
 			}
 			for i := range repoResp.Repositories {
+				assets = append(assets, ecrRepoToAsset(repoResp.Repositories[i], region, a.profile))
 				repoName := repoResp.Repositories[i].RepositoryName
+				repoUrl := *repoResp.Repositories[i].RepositoryUri
 				imageResp, err := svc.DescribeImages(ctx, &ecr.DescribeImagesInput{
 					RepositoryName: repoName,
 				})
 				if err != nil {
-					return imgs, nil
+					return assets, nil
```

Comment:
Not sure whether my thought is valid:

When we get multiple repos from AWS and an error occurs on the first repo, we will not try to query the other repos. Perhaps this can happen when limited credentials are provided or something like this. Not sure how much of an edge-case this is. My thoughts also only apply to private repos. Don't whether this could also happen in a different situation to public repos. Perhaps something for v2 of the ECR support?

Reply:
yes that's a great point!